projects/br_scs/src/Mirza/BusinessRegistry/Types.hs

```diff
@@ -61,41 +59,59 @@ import           GHC.Stack                              (CallStack)
 -- Context Types
 -- *****************************************************************************
 
-data BRContext = BRContext
+type BRContextMinimal  = BRContextGeneric () ()
+type BRContextComplete = BRContextGeneric Audience JWK
+
+brContextMinimal :: EnvType -> Pool Connection -> K.LogEnv -> K.LogContexts -> K.Namespace -> BRContextMinimal
+brContextMinimal a b c d e = BRContextGeneric a b c d e () ()
+
+brContextComplete :: BRContextMinimal -> Audience -> JWK-> BRContextComplete
+brContextComplete (BRContextGeneric a b c d e () ()) f g = BRContextGeneric a b c d e f g
+
+data BRContextGeneric audienceType publicKeyType = BRContextGeneric
```

Comment:
In what cases can we expect the `audienceType` to be different to `Audience`?

Reply:
When it is unit `()`, the point is that sometimes we want the full "record", but sometimes I want the same datatype minus some of the fields. Since haskell doesn't support adding to or removing fields from a datatype have used this design to support adding fields (or removing fields) from a type.

Let me know if that doesn't make sense and I will try to explain in a different way.

Reply:
Oh okay, but if the question is "Would there be an Audience or not?", then is it not better to have audience as a `Maybe`?

Reply:
With () you can gaurentee at the type level (compile time) whether audience is present in the type or not, where as maybe defers that to runtime, hence you can detect the logic errors of whether the audience is present in locations where it is required while you are developing via compiler vs finding out the logic error when someone hits it in a runtime condition.